src/diffusers/models/vae.py

```diff
@@ -1,12 +1,55 @@
+from dataclasses import dataclass
+
 import numpy as np
 import torch
 import torch.nn as nn
 
 from ..configuration_utils import ConfigMixin, register_to_config
 from ..modeling_utils import ModelMixin
+from ..utils import ModelOutput
 from .unet_blocks import UNetMidBlock2D, get_down_block, get_up_block
 
 
+@dataclass
+class DecoderOutput(ModelOutput):
+    """
+    Output of decoding method.
+
+    Args:
+        sample (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Decoded output sample of the model. Output of the last layer of the model.
+    """
+
+    sample: torch.FloatTensor = None
+
+
+@dataclass
+class VQEncoderOutput(ModelOutput):
+    """
+    Output of VQModel encoding method.
+
+    Args:
+        latents (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Encoded output sample of the model. Output of the last layer of the model.
+    """
+
+    latents: torch.FloatTensor = None
+
+
+@dataclass
+class AutoEncoderKLOutput(ModelOutput):
+    """
+    Output of VQModel encoding method.
+
+    Args:
+        latent_dist (`DiagonalGaussianDistribution`):
+            Encoded outputs of `Encoder` represented as the mean and logvar of `DiagonalGaussianDistribution`.
+            `DiagonalGaussianDistribution` allows for sampling latents from the distribution.
+    """
+
+    latent_dist: torch.FloatTensor = None
```

Comment:
This should be of type `DiagonalGaussianDistribution` as the comment says.

The problem is that `DiagonalGaussianDistribution` is defined later in the file. Solutions:
- Move this down. It would be out of place with respect to the other output dataclasses.
- Declare `class DiagonalGaussianDistribution: pass` before, and comment it's a "forward" declaration.
- Use `object` as the type, with a comment.

Reply:
Using the string version instead as suggested here: https://peps.python.org/pep-0484/#forward-references

Reply:
Ohh, nice!